src/main/java/rx/internal/operators/OperatorDistinctUntilChanged.java

```diff
@@ -68,7 +83,16 @@ public void onNext(T t) {
                 previousKey = key;
                 
                 if (hasPrevious) {
-                    if (!(currentKey == key || (key != null && key.equals(currentKey)))) {
+                    Boolean comparison;
+                    
+                    try {
+                        comparison = comparator.call(currentKey, key);
+                    } catch (Throwable e) {
+                        Exceptions.throwOrReport(e, child, key);
+                        return;
+                    }
+                    
+                    if (!comparison) {
```

Comment:
There's a subtle NPE potential here. Not sure if you want to explicitly guard against it by a misbehaving `comparator` implementation.